File: AXServer/Sources/CodeAlpha/websocket/messages/types/Event.swift
import Foundation
import Vapor

enum EventType: String, Codable {
  case AppFocusState
  case XCodeEditorContent
  case XCodeFocusStatus
  case XCodeFocusStatusChange
}

struct Event<T: Codable>: Codable {
  let eventType: EventType
  let payload: T

  enum CodingKeys: String, CodingKey {
    case eventType = "event"
    case payload
  }
}

extension ByteBuffer {
  func decodeEventMessage<T: Codable>(_: T.Type) -> Event<T>? {
    let decoder = JSONDecoder()
    decoder.keyDecodingStrategy = .convertFromSnakeCase
    return try? decoder.decode(Event<T>.self, from: self)
  }
}
  
 
 

File: AXServer.Companion/src/ax_messages/types/event.rs
use serde::{Deserialize, Serialize};

pub use super::super::models::{
    AppFocusState, XCodeEditorContent, XCodeFocusStatus, XCodeFocusStatusChange,
};

#[derive(Serialize, Deserialize, Debug)]
#[serde(tag = "event", content = "payload")]
pub enum Event {
    AppFocusState(AppFocusState),
    XCodeEditorContent(XCodeEditorContent),
    XCodeFocusStatus(XCodeFocusStatus),
    XCodeFocusStatusChange(XCodeFocusStatusChange),
    None,
}
